spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -480,6 +480,13 @@ def on_remote_client_available(self):
     def on_main_interpreter_available(self):
         main_interpreter = self.get_plugin(Plugins.MainInterpreter)
         main_interpreter.sig_environments_updated.connect(self._update_envs)
+        mi_container = main_interpreter.get_container()
+
+        pythonenv_status = self.get_widget().pythonenv_status
+
+        # connect signals
+        pythonenv_status.sig_open_preferences_requested.connect(
+            mi_container.sig_open_preferences_requested)
```

Comment:
There's no need to declare a method in the Main interpreter container for this because it makes the code harder to follow. Instead, we  can add it here to `IPythonConsoleWidget`.

For that, please remove these changes and connect `pythonenv_status.sig_open_preferences_requested` in line **391** below:

https://github.com/spyder-ide/spyder/blob/66d9f6a62f2709f1dc43a37a11e57c463913ece8/spyder/plugins/ipythonconsole/widgets/main_widget.py#L388-L391

like this:

```python
pythonenv_status.sig_open_preferences_requested.connect(
    self._open_interpreter_preferences
)
```